File: app/src/main/java/com/originb/inkwisenote2/modules/noterelation/data/TermNode.java
package com.originb.inkwisenote2.modules.noterelation.data;

import java.util.HashSet;
import java.util.Set;

public class TermNode {
    String termId;
    Set<String> documentNodes; // Document ID to DocumentNode
    Double idfScore;

    public TermNode(String termId) {
        this.termId = termId;
        this.documentNodes = new HashSet<>();
        idfScore = 0.0;
    }
}

File: androidApp/src/main/java/com/originb/inkwisenote2/modules/noterelation/data/TermNode.kt
package com.originb.inkwisenote2.modules.noterelation.data

class TermNode(var termId: String?) {
    var documentNodes: MutableSet<String?>? // Document ID to DocumentNode
    var idfScore: Double? = 0.0

    init {
        this.documentNodes = HashSet<String?>()
    }
}



